src/dymaptic.GeoBlazor.Core/Components/MapComponent.razor.cs

```diff
@@ -240,6 +271,421 @@ protected virtual async Task RenderView(bool forceRender = false)
+        bool once, bool initial)
+    {
+        string typeName = GetType().Name;
+        targetName ??= typeName.Contains("View") ? "view" : typeName.ToLowerFirstChar();
+
+        if (!watchExpression.Contains(targetName))
+        {
+            throw new UnMatchedTargetNameException(targetName, watchExpression);
+        }
+        IJSObjectReference? jsRef = 
+            await JsModule!.InvokeAsync<IJSObjectReference?>("addReactiveWatcher", Id, 
+                targetName, watchExpression, once, initial);
+
+        if (jsRef != null)
+        {
+            _watchers[watchExpression] = (handler, jsRef);
+        }
+    }
+
+    /// <summary>
+    ///     Removes the tracker on a particular expression.
+    /// </summary>
+    /// <param name="watchExpression">
+    ///     The expression to stop tracking.
+    /// </param>
+    public async Task RemoveReactiveWatcher(string watchExpression)
+    {
+        if (_watchers.ContainsKey(watchExpression))
+        {
+            IJSObjectReference jsRef = _watchers[watchExpression].JsObjRef;
+            await jsRef.InvokeVoidAsync("remove");
+            _watchers.Remove(watchExpression);
+        }
+    }
+
+#pragma warning disable CS1574
+    /// <summary>
+    ///     JS-Invokable method that is triggered by the reactiveUtils watchers. This method will dynamically trigger handlers passed to <see cref="AddReactiveWatcher"/>
+    /// </summary>
+    /// <param name="watchExpression">
+    ///     The tracked expression that was triggered.
+    /// </param>
+    /// <param name="value">
+    ///     The return value of the watcher callback.
+    /// </param>
+#pragma warning restore CS1574
+    [JSInvokable]
+    public void OnReactiveWatcherUpdate(string watchExpression, JsonElement? value)
+    {
+        Delegate handler = _watchers[watchExpression].Handler;
+        Type returnType = handler.Method.GetParameters()[0].ParameterType;
+        object? typedValue = null;
+
+        if (value.HasValue)
+        {
+            string stringValue = value.Value.ToString();
+            var options = new JsonSerializerOptions() { PropertyNameCaseInsensitive = true };
+            typedValue = value.Value.ValueKind switch
+            {
+                JsonValueKind.Object => value.Value.Deserialize(returnType, options),
+                JsonValueKind.Array => value.Value.Deserialize(returnType, options),
+                JsonValueKind.False => false,
+                JsonValueKind.True => true,
+                JsonValueKind.Number => Convert.ChangeType(stringValue, returnType),
+                JsonValueKind.String => stringValue,
+                _ => typedValue
+            };
+        }
+
+        handler.DynamicInvoke(typedValue);
+    }
+
+    
+    /// <summary>
+    ///     Tracks any properties accessed in the <code>listenExpression</code> and calls the callback when any of them change.
```

Comment:
what's the difference between Listener and Watcher? (is this `on` vs `watch` in JS?)

Reply:
Yes, exactly. On/Listener is for events, Watch is for property values.